data-structures-queues/take-next-smallest.js

```diff
@@ -0,0 +1,18 @@
+/* exported takeNextSmallest */
+
+function takeNextSmallest(queue) {
+  if (queue.peek === undefined) {
+    return undefined;
+  }
```

Comment:
I believe you meant to call this. If the tests are passing anyway, this conditional is probably unnecessary.